tensorflow/python/kernel_tests/diag_op_test.py

```diff
@@ -451,11 +499,23 @@ def testRankSixFloatTensor(self):
     self.diagPartOp(x, np.float32, expected_ans)
     self.diagPartOp(x, np.float64, expected_ans)
 
+  def testRankEightComplexTensor(self):
+    x = np.random.rand(2, 2, 2, 3, 2, 2, 2, 3)
```

Comment:
Consider doing a loop in this test so we test multiple cases in a single test run.

Reply:
It's a little hard to clean the code in single test run. Because `numpy.diag` do not support multi-dimension tensor, we must extract the corresponding diagonal elements by hand, so I may think it's more clearer to unroll the test cases in this way.